src/Microsoft.OData.Edm/Schema/EdmAnnotationsTarget.cs

```diff
@@ -0,0 +1,55 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmAnnotationsTarget.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Edm.Vocabularies;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Represents an EDM annotations target.
+    /// </summary>
+    public class EdmAnnotationsTarget : IEdmAnnotationsTarget
+    {
+        private IEnumerable<IEdmElement> targetSegments;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmAnnotationsTarget"/> class.
+        /// </summary>
+        /// <param name="targetSegments">Target segments.</param>
+        public EdmAnnotationsTarget(IEnumerable<IEdmElement> targetSegments)
+        {
+            EdmUtil.CheckArgumentNull(targetSegments, nameof(targetSegments));
+
+            // Validate that the first element is Container.
+            if (!(targetSegments.First() is IEdmEntityContainer))
+            {
+                throw new InvalidOperationException(Strings.EdmAnnotationsTarget_FirstSegmentMustBeIEdmEntityContainer);
+            }
+
+            this.targetSegments = targetSegments;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmAnnotationsTarget"/> class.
+        /// </summary>
+        /// <param name="targetSegments">Target segments.</param>
+        public EdmAnnotationsTarget(params IEdmElement[] targetSegments)
+            :this((IEnumerable<IEdmElement>)targetSegments)

```

Comment:
```suggestion
            :this(targetSegments.AsEnumerable())

```

This avoids an unnecessary cast